src/server/routes/entity/edition.ts

```diff
@@ -158,37 +158,33 @@ router.get(
 	'/create', auth.isAuthenticated, auth.isAuthorized(ENTITY_EDITOR), middleware.loadIdentifierTypes,
 	middleware.loadEditionStatuses, middleware.loadEditionFormats,
 	middleware.loadLanguages, middleware.loadRelationshipTypes,
-	(req:PassportRequest, res, next) => {
+	async (req:PassportRequest, res, next) => {
 		const {EditionGroup, Publisher, Work, Author} = req.app.locals.orm;
-		const propsPromise = generateEntityProps(
+		const propsObject = generateEntityProps(
 			'edition', req, res, {}
 		);
+		async function fetchData(entity, key) {
```

Comment:
Refactoring is good !

We do have a utility function in the ORM that does the same thing while also making sure to follow merged BBIDs:
https://github.com/metabrainz/bookbrainz-data-js/blob/master/src/func/entity.ts#L199-L212

So you should be able to use `await orm.func.entity.getEntity(orm, 'Work', bbid, ['defaultAlias']);` directly instead of rolling out this fetchData utility.

There is a small difference in that the fetch option `require` is set to true instead of false, but I think in this context it does make sense to ensure the entity we're trying to link does exist.

I think you might also have to move the try…catch block around all these `if (req.query.xyz)` so that error are caught (especially with `require:true` as stated above)